sycl/tools/sycl-xocc/bin/sycl-xocc

```diff
@@ -13,56 +13,77 @@
 # unaware of). So no kernel names and no idea how many xocc commands you'd need
 # to generate per file (no idea how many kernels are in a file).
 #
-# This works around that by having the driver create an opt pass that generates
+# This works around that by using an opt (kernelNameGen) pass that generates
 # an intermediate file with the needed information that we eat up and can then
 # loop over each kernel in a file. It's simple at the moment just kernel names,
 # but could expand in the future to include optimization information for each
 # kernel.
 #
 # Arguments:
 # $1 == xocc's real shell invocation that invokes the xocc compiler
-# $2 == -c or -l for the current xocc stage
-# $3 == data file contianing all of our lovely kernel names that we wish to
-#       compile
-# $4 == our .xpirbc input file or the name of our linker output file
-# $5 == temporary file directory to place intermediate .xo files or to retrieve
-#       intermediate .xo files for linking when -l speciifed
+# $2 == path to the clang driver that's executing the shell script (e.g. 
+#        your compiled Clang's build/bin)
+# $3 == The name of the source file without file extensions or temporary 
+#       manglings, mostly used to generate some intermediate file names
+# $4 == The initial input file name, currently a .o file that's actually a .bc 
+#       file in disguise       
+# $5 == The systems temporary directory where we'll deposit some intermediate 
+#       files
+# $6 == The name we wish to give our final output .xcl binary, important that it
+#       is consistent with what the driver expects it to be. 
 
 # TODO: Pass an argument through the driver to set options like XILINX_PLATFORM, 
 # the environment variables are a quick workaround to swap boards. 
-
 usage() { echo xocc_compile: error: $2 >&2; exit $1; }
 
-[ $# -eq 0      ] && usage 1 "no xocc shell or bin"
-[ $# -eq 0      ] && usage 2 "no compilation option specified"
-[ $# -eq 0      ] && usage 3 "no data file"
+[ $# -eq 0      ] && usage 1 "no sdx bin directory containing xocc"
+[ $# -eq 0      ] && usage 2 "no driver path specified"
+[ $# -eq 0      ] && usage 3 "no source file name"
+[ $# -eq 0      ] && usage 4 "no input file name"
+[ $# -eq 0      ] && usage 5 "no temporary directory specified"
+[ $# -eq 0      ] && usage 6 "no final output file specified"
+
+# Hardcoded path from the bin directory that xocc is stored in, works for now 
+# but a future TODO might be to allow this to be optionally specified as an 
+# ENV variable 
+XOCC_LIB_SPIR=$1/../lnx64/lib/libspir64-39-hls.bc
```

Comment:
Put some "" everywhere, such as `"$1/../lnx64/lib/libspir64-39-hls.bc"`, just in case there are spaces in the arguments :-)